deploy/operator/ztp/add_day2_remote_nodes.sh

```diff
@@ -44,6 +52,33 @@ function remote_done_agents() {
 
 export -f remote_done_agents
 
+# If we are performing late binding then each of the agents needs to have the correct clusterDeploymentRef applied.
+# This needs to happen after the agents are available. They cannot move to "done" until the clusterDeploymentRef is applied.
+if [ -z "${DAY2_LATE_BINDING}" ] ; then
+  clusterDeploymentName=${LATE_BINDING_ASSISTED_CLUSTER_DEPLOYMENT_NAME}
+
+  # Generate a patch to assign the correct cluster name.
+  agentPatch=$(cat <<PATCH
+  {
+    "spec":
+    {
+      "clusterDeploymentName":{
+          "name":"${clusterDeploymentName}",
+          "namespace":"${SPOKE_NAMESPACE}"
+      }
+    }
+  }
+PATCH
+  )
+
+  # Apply this patch to our Day2 agents.
+  for agentHostName in $(cat ${REMOTE_BAREMETALHOSTS_FILE}  | jq -r '.[].name') ; do
```

Comment:
Double quotes around `${REMOTE_BAREMETALHOSTS_FILE}` to not break on paths containing spaces